WalletWasabi.Fluent/ViewModels/Wallets/Buy/Workflows/Workflow.cs

```diff
@@ -55,11 +57,29 @@ protected async Task ExecuteStepAsync(IWorkflowStep step)
 				await step.ExecuteAsync();
 				break;
 			}
-			catch (Exception ex)
+			catch (Exception ex) when (ex is not OperationCanceledException)
 			{
 				step.Reset();
-				Logger.LogError($"An error occurred trying to execute Step '{step.GetType().Name}' in Workflow '{GetType().Name}'", ex);
-				OnStepError.SafeInvoke(this, ex);
+
+				if (step.IsInteractive)
+				{
+					OnStepError.SafeInvoke(this, ex);
+					Logger.LogError($"An error occurred trying to execute Step '{step.GetType().Name}' in Workflow '{GetType().Name}'", ex);
+				}
+
+				if (!step.IsInteractive)
+				{
+					if (lastException?.Message != ex.Message)
+					{
+						lastException = ex;
+						Logger.LogError($"An error occurred trying to execute Step '{step.GetType().Name}' in Workflow '{GetType().Name}'", ex);
+					}
+
+					var isBusy = step.IsBusy;
+					step.IsBusy = true;
+					await Task.Delay(3000, token);
```

Comment:
Please put a comment on "why" you must hack.

Reply:
done